backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/cgen/CBridgeGen.kt

```diff
@@ -671,6 +669,9 @@ private sealed class TypeLocation(val element: IrElement) {
     class BlockReturnValue(val blockLocation: TypeLocation) : TypeLocation(blockLocation.element)
 }
 
+private val TypeLocation.shallUseCStrings: Boolean
+    get() = this is TypeLocation.FunctionArgument || this is TypeLocation.FunctionPointerParameter
```

Comment:
This is wrong, see `interop_objc_smoke` test.
Also this will likely break support for C function types with `NSString*`.